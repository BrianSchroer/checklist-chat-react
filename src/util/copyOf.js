export default original => Object.assign({}, original);
