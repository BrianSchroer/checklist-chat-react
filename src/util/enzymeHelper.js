const enzymeHelper = class {

    static nestedFind(wrapper, selector) {
        const selectors = selector.split('>').map(node => node.trim());
        let parent = wrapper;
        let found = [];

        for (let i = 0; i < selectors.length; i++) {
            found = parent.find(selectors[i]);
            if (!found.length) {
                break;
            }
        }

        return found;
    }

    static findSingle(wrapper, selector) {
        const findArray = enzymeHelper.nestedFind(wrapper, selector);

        if (findArray.length != 1) {
            throw(`expected 1 find for selector "${selector}", but found ${findArray.length}.`);
        }

        return findArray.first();
    }
};

export default enzymeHelper;
