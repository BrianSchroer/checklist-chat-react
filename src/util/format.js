import moment from 'moment';

export default class {
    static time(input = '') {

        const isoDateFormatRegex = /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}([+-]\d{2}:\d{2})?/g;
        let formatted = input;

        if (input && input.match(isoDateFormatRegex)) {
            formatted = this.dateTimeString(input, 'h:mm a');
        }

        return formatted;
    }

    static dateTimeString(input, formatString) {
        let formatted = input;

        if (input) {
            const testMoment = moment(input);

            if (testMoment.isValid()) {
                const testFormat = testMoment.format(formatString);
                if (testFormat != 'Invalid date') {
                    formatted = testFormat;
                }
            }
        }

        return formatted;
    }
}
