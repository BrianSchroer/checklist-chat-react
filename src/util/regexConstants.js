export const CSS_PATH = /(\.css)$/;
export const EOT_PATH = /\.eot(\?v=\d+\.\d+\.\d+)?$/;
export const JS_PATH = /\.js$/;
export const SVG_PATH = /\.svg(\?v=\d+\.\d+\.\d+)?$/;
export const TTF_PATH = /\.ttf(\?v=\d+\.\d+\.\d+)?$/;
export const WOFF_OR_WOFF2_PATH = /\.(woff|woff2)(\?v=\d+\.\d+\.\d+)?$/;
