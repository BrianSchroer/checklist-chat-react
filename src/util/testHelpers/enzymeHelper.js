import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

export default class EnzymeHelper {
  /**
   * Create new EnzymeHelper instance
   * @param {*} component - React component
   */
  constructor(component) {
    this.component = component.type;
    this.props = component.props;
    this.adjustProps = (props, propsAdjustments) =>
      Object.assign({}, props, propsAdjustments);
    this.wrapper = null;
  }

  /**
   * Fluent syntax helper to specify a callback function to be called by the test() function
   * to assign propsAdjustments to the props passed that were passed via the constructor.
   * (Defaults to assigning top-level props only.)
   */
  withPropsAdjuster = propsAdjuster => {
    this.adjustProps = propsAdjuster;
    return this;
  };

  /**
   * Creates a new props object by calling the propsAdjuster function to apply
   * propsAdjustments (if any) to the props passed to the constructor, then
   * returns a "shallowWrapper" of the component with the specified props.
   */
  shallow = propsAdjustments => {
    const Component = this.component;
    const props = this.adjustProps(this.props, propsAdjustments);

    return (this.wrapper = shallow(<Component {...props} />));
  };

  /**
   * This method wraps the enzyme find method with the ability to perform
   * "nested" finds (e.g. "div.main > div.sub > p.instructions") for the shallow "wrapper"
   * created/returned by a previous call to the "shallow" function.
   *
   * @static
   * @param {any} selector for the search
   * @param {string} higherOrderComponent = optional higher-order component (e.g. "Connect")
   *      that selector may be wrapped with.
   * @returns array of found results
   */
  find = (selector, higherOrderComponent) => {
    return this.findIn(this.wrapper, selector, higherOrderComponent);
  };

  /**
   * This method wraps the enzyme find method with the ability to perform
   * "nested" finds (e.g. "div.main > div.sub > p.instructions").
   *
   * @param {any} wrapper to be searched
   * @param {any} selector for the search
   * @param {string} higherOrderComponent = optional higher-order component (e.g. "Connect")
   *      that selector may be wrapped with.
   * @returns array of found results
   */
  findIn = (wrapper, selector, higherOrderComponent) => {
    const selectors = selector.split('>').map(node => node.trim());
    let parent = wrapper;
    let found = [];

    for (let i = 0; i < selectors.length; i++) {
      found = parent.find(selectors[i]);

      if (found.length == 0) {
        if (higherOrderComponent) {
          found = parent.find(`${higherOrderComponent}(${selectors[i]})`);
        }

        if (found.length == 0) {
          break;
        }
      }
    }

    return found;
  };

  /**
   * Call "find" function and assert the expected match count.
   *
   * @param {number} expectedCount
   * @param {any} selector for the search (can be "nested"; e.g. "div > span")
   * @param {string} higherOrderComponent = optional higher-order component (e.g. "Connect")
   *      that selector may be wrapped with.
   * @returns array of found results
   */
  assertFindCount = (expectedCount, selector, higherOrderComponent) => {
    return this.assertFindCountIn(
      this.wrapper,
      expectedCount,
      selector,
      higherOrderComponent
    );
  };

  /**
   * Call "findIn" function and assert the expected match count.
   *
   * @param {any} wrapper to be searched
   * @param {number} expectedCount
   * @param {any} selector for the search (can be "nested"; e.g. "div > span")
   * @param {string} higherOrderComponent = optional higher-order component (e.g. "Connect")
   *      that selector may be wrapped with.
   * @returns array of found results
   */
  assertFindCountIn = (
    wrapper,
    expectedCount,
    selector,
    higherOrderComponent
  ) => {
    const found = this.findIn(wrapper, selector, higherOrderComponent);

    if (found.length != expectedCount) {
      const errorMessage = [
        `Expected ${expectedCount} element(s) for selector "${selector}"` +
          `, but found ${found.length}.`,
        '',
        this.getDebugHtml(wrapper),
        ''
      ].join('\n');

      throw new Error(errorMessage);
    }

    return found;
  };

  /**
   * Call "find" method and assert that no matches are found.
   *
   * @param {any} selector for the search (can be "nested"; e.g. "div > span")
   * @param {string} higherOrderComponent = optional higher-order component (e.g. "Connect")
   *      that selector may be wrapped with.
   */
  assertNoMatch = (selector, higherOrderComponent) => {
    this.assertNoMatchIn(this.wrapper, selector, higherOrderComponent);
  };

  /**
   * Call "findIn" method and assert that no matches are found.
   *
   * @param {any} wrapper to be searched
   * @param {any} selector for the search (can be "nested"; e.g. "div > span")
   * @param {string} higherOrderComponent = optional higher-order component (e.g. "Connect")
   *      that selector may be wrapped with.
   */
  assertNoMatchIn = (wrapper, selector, higherOrderComponent) => {
    this.assertFindCountIn(wrapper, 0, selector, higherOrderComponent);
  };

  /**
   * Call "find" function and assert that one and only one match is found.
   *
   * @param {any} selector for the search (can be "nested"; e.g. "div > span")
   * @param {string} higherOrderComponent = optional higher-order component (e.g. "Connect")
   *      that selector may be wrapped with.
   * @returns the found node
   */
  findSingle = (selector, higherOrderComponent) => {
    return this.findSingleIn(this.wrapper, selector, higherOrderComponent);
  };

  /**
   * Call "findIn" function and assert that one and only one match is found.
   *
   * @param {any} wrapper to be searched
   * @param {any} selector for the search (can be "nested"; e.g. "div > span")
   * @param {string} higherOrderComponent = optional higher-order component (e.g. "Connect")
   *      that selector may be wrapped with.
   * @returns the found node
   */
  findSingleIn = (wrapper, selector, higherOrderComponent) => {
    const found = this.assertFindCountIn(
      wrapper,
      1,
      selector,
      higherOrderComponent
    );

    return found.first();
  };

  /**
   * Returns an HTML-like string of the shallow wrapper for debugging purposes. Useful to
   * log to the console when tests are not passing when you expect them to.
   *
   * @param {any} wrapper (if not passed, uses the wrapper retrieved/returned by preceding "shallow" function call)
   * @returns the resulting string
   */
  getDebugHtml = wrapper => (wrapper || this.wrapper).debug();

  /**
   * Logs an HTML-like string of the wrapper for debugging purposes. Useful to
   * log to the console when tests are not passing when you expect them to.
   *
   * @param {any} wrapper (if not passed, uses the wrapper retrieved/returned by preceding "shallow" function call)
   * @returns the resulting string
   */
  logDebugHtml = wrapper =>
    console.log(this.getDebugHtml(wrapper || this.wrapper)); // eslint-disable-line no-console
}
