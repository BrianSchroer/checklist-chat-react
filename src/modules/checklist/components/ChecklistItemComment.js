import React from 'react';
import PropTypes from 'prop-types';
import {format} from '../../../util';
import {ChatMessageMarkdown} from '../../chat';

const ChecklistItemComment = ({chatMessage}) => {
    return (
        <tr className="checklist-item-comment-row small text-muted">
            <td>&nbsp;</td>
            <td>&nbsp;</td>
            <td className="timestamp">{format.time(chatMessage.timeStamp)}</td>
            <td colSpan="4">
                <strong>{`${chatMessage.userName}: `}</strong>
                <ChatMessageMarkdown source={chatMessage.text} className="chat-message-text" />
            </td>
        </tr>
    );
};

ChecklistItemComment.propTypes = {
    chatMessage: PropTypes.object.isRequired
};

export default ChecklistItemComment;
