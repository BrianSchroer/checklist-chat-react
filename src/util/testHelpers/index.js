import {shallow, mount} from 'enzyme';
import enzymeHelper from './enzymeHelper';

export {
    shallow as shallow,
    mount as mount,
    enzymeHelper as enzymeHelper
};
