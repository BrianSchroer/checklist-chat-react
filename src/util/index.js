import format from './format';
import uiHelper from './uiHelper';
import validationHelper from './validationHelper';

export {
    format as format,
    uiHelper as uiHelper,
    validationHelper as validationHelper
};
