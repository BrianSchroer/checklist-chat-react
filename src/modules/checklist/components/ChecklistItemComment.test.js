import React from 'react';
import {snapshotHelper} from '../../../util/testHelpers';
import ChecklistItemComment from './ChecklistItemComment';

describe('ChecklistItemComment', () => {
    it('should render correctly', () => {
        const props = {
            chatMessage: {
                timeStamp: '2016-12-08T14:57:10.222Z',
                userName: 'test userName',
                text: 'test text'
            }
        };

        snapshotHelper.assertMatch(<ChecklistItemComment {...props} />);
    });
});
