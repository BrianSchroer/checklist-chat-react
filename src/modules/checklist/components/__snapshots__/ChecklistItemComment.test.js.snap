// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChecklistItemComment should render correctly 1`] = `
<tr
  className="checklist-item-comment-row small text-muted"
>
  <td>
     
  </td>
  <td>
     
  </td>
  <td
    className="timestamp"
  >
    8:57 am
  </td>
  <td
    colSpan="4"
  >
    <strong>
      test userName: 
    </strong>
    <span
      className="chat-message-text"
    >
      test text
    </span>
  </td>
</tr>
`;
